src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/JavaScriptTests.cs

```diff
@@ -223,5 +223,99 @@ public static IEnumerable<object> ToEnumerable(this JSObject iterrator)
                 nextResult?.Dispose();
             }
         }
+
+        private static JSObject SetupListenerTest (string prefix) {
+            Runtime.InvokeJS($"globalThis.{prefix} = {{" + @"
+    listeners: [],
+    addEventListener: function (name, listener, options) {
+        if (name === 'throwError')
+            throw new Error('throwError throwing');
+        for (var i = 0; i < this.listeners.length; i++) {
+            var item = this.listeners[i];
+            if (item[0] !== name)
+                continue;
+            if (item[1] === listener)
+                return;
+        }        
+        this.listeners.push([name, listener, options || null]);
+    },
+    removeEventListener: function (name, listener, capture) {
+        for (var i = 0; i < this.listeners.length; i++) {
+            var item = this.listeners[i];
+            if (item[0] !== name)
+                continue;
+            if (item[1] !== listener)
+                continue;
+            if (item[2] === null)
+                continue;
+            this.listeners.splice(i, 1);
+            return;
+        }
+    },
+    fireEvent: function (name, evt) {
```

Comment:
Is there a test using the `evt`?

Reply:
I could add one, but it's not really necessary since this API doesn't actually touch the event dispatch part